### PR TITLE
refactor cmd/config to internal

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
@monopole 
Finally getting around this this

File: cmd/config/configcobra/cmds.go [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+package configcobra
 
 import (
 	"strings"
 
 	"github.com/spf13/cobra"
-	"sigs.k8s.io/kustomize/cmd/config/cmd"
-	"sigs.k8s.io/kustomize/cmd/config/cmddocs/api"
-	"sigs.k8s.io/kustomize/cmd/config/cmddocs/tutorials"
+	"sigs.k8s.io/kustomize/cmd/config/internal/cmd"
```
Comment (monopole) [inline cmd/config/configcobra/cmds.go]: while you're here and in the mood -

how about renaming `internal/cmd` to `internal/commands`, as is done here 
https://github.com/kubernetes-sigs/kustomize/tree/master/kustomize

This reserves `cmd` for top level un-importable mains, per https://github.com/golang-standards/project-layout#cmd

everything else looks good.  could pull this first if you like

Comment (pwittrock) [reply cmd/config/configcobra/cmds.go]: I'll update it now.

**Pull request merged**